Add rust targets to README

When new clone and build you get
```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-unknown-linux-musl` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-unknown-linux-musl`

For more information about this error, try `rustc --explain E0463`.
error: could not compile `scopeguard` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `cfg-if` (lib) due to 1 previous error
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-linux-musl` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-unknown-linux-musl`

error: could not compile `log` (lib) due to 1 previous error
error: could not compile `memchr` (lib) due to 1 previous error
```

Maybe some README additions?

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,3 +1,2 @@
-[build]
+[target."x86_64-unknown-linux-musl"]
 rustflags = ["-C", "target-feature=+crt-static"]
-target = "x86_64-unknown-linux-musl"

